src/interfaces/resources/resourcesAddViewUI.py

```diff
@@ -0,0 +1,411 @@
+"""
+
+    This file provides a pure GUI interface for resources.
+    This component provides a workflow style interface to add current machine to resources pool.
+
+"""
+
+from abc import ABCMeta, abstractmethod
+
+from PyQt5.QtWidgets import QFrame, QLineEdit, QLabel, QRadioButton
+from PyQt5.QtCore import pyqtSignal, Qt
+
+from ..widgets import (Frame, SectionTitleFrame, ConfigFrame, TabsInputFrame,
+                        ViewButton, Label, PriceBox,
+                        HorizontalLayout, VerticalLayout, StackLayout,
+                        HorizontalSpacer, VerticalSpacer)
+
+from ..util import get_children
+from ..stylesheet import resources_add_view_style
+
+class ResourcesAddViewUI(Frame):
+
+    # metaclass for defining abstract base classes
+    __metaclass__ = ABCMeta
```

Comment:
What's up with this?

Reply:
That's a way to define the class that contain abstract function. A reference can be found [here](https://docs.python.org/2/library/abc.html#abc.ABCMeta.register).
There's some function that I want to pre-defined as the ui layer i.e. `on_xxxx_clicked()`, then I can connect the function with the widget during init period. Usually we can just defined it and leave it that, such as :
```python
def on_xxxx_clicked(self):
   pass
```
However, I do want some on clicked that would be a "required" that have to be implemented in the outer field. Therefore, I chose `@abstractmethod` decorator.

Reply:
Is there a difference between defining `ResourcesAddViewUI` as a subclass of `ABCMeta` and defining `__metaclass__ = ABCMeta` as a class variable?

Also, can you please add that reference to the comment!

Reply:
Good question, and Yes! By reference this [section](https://docs.python.org/2/library/abc.html#abc.abstractmethod), Unlike Java abstract methods, these abstract methods may have an implementation. This implementation can be called via the super() mechanism from the class that overrides it. This could be useful as an end-point for a super-call in a framework that uses cooperative multiple-inheritance.

Reply:
And if you do some search, you can see the people recommand to do that, in case to avoid the potential conflict problem.

Reply:
Okay, I added the reference to the comment above this line.